File: kq-projection-dotnet/kq-projection/Common/Models/Jigsaw/JigsawPuzzlePiece.cs
using System.Text.Json.Serialization;

namespace KQ_Projection.Common.Models.Jigsaw;
public class JigsawPuzzlePiece
{
    [JsonPropertyName("pieceId")]
    public Guid pieceId { get; set; }

    [JsonPropertyName("column")]
    private int column {  get; set; }

    [JsonPropertyName("row")]
    private int row { get; set; }

    [JsonPropertyName("pieceData")]
    public byte[] pieceData { get; set; }
}

File: java-apps/kq-common/src/main/java/com/kafkaquest/kq/common/util/models/events/other/JigsawPuzzlePiece.java
package com.kafkaquest.kq.common.util.models.events.other;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class JigsawPuzzlePiece {

    private UUID pieceId;
    private int column;
    private int row;
    private byte[] pieceData;
}
